### DISCUSSION TITLE
Escaping ampersand character with following semicolon

### DISCUSSION THREAD

**Discussion by Quty** [oozcitak/xmlbuilder2#169] [Q&A] [answered]
Following code

```js
const xmlbuilder = require('xmlbuilder2');

const xml = xmlbuilder.create({ encoding: 'UTF-8' })
    .ele('root').txt('5&ABC;')
    .end({ prettyPrint: true });

console.log(xml);
```

outputs

```xml
<?xml version="1.0" encoding="UTF-8"?>
<root>5&ABC;</root>
```

Is it expected that `#txt` method doesn't escapes `&` character in `5&ABC;`?

**Comment by mglanville** [comment, marked as answer]
I think the problem is the way that xmlbuilder2 tries to avoid double-encoding XML escape sequences. Your string is being treated as an XML entity by [nonEntityAmpersandRegex](https://github.com/oozcitak/xmlbuilder2/blob/c47f5b95c6e48a8bf56649321790eb5bfe035df4/src/constants.ts#L1C24-L1C24) so the ampersand doesn't get replaced.

I'm currently hitting a problem with the same code, but that's because I wan't the ampersand in the text `b&amp;w` to be encoded and xmlbuilder2 seems to have been deliberately built to not do that.

**Reply by Quty** [reply]
This appears to be double-encoding protection.

In my opinion, the method should either accept raw data and encode all invalid characters, or accept prepared data and not perform any conversions.

Now I am forced to use third party tools to encode the text before passing it to `#txt` method.
